diff --git a/gib/ircbot.py b/gib/ircbot.py
index 5c43b03..4c38649 100644
--- a/gib/ircbot.py
+++ b/gib/ircbot.py
@@ -22,6 +22,8 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
+import random
+
 from twisted.words.protocols import irc
 from twisted.internet import reactor, protocol, task
 from twisted.python import log
@@ -55,14 +57,10 @@ class GoogleCodeIRCBot ( irc.IRCClient ):
 	def connectionMade ( self ):
 		irc.IRCClient.connectionMade( self )
 		log.msg( self.nickname + ": Connected" )
-		if self.logger:
-			self.logger.connected()
 
 	def connectionLost( self, reason ):
 		irc.IRCClient.connectionLost( self, reason )
 		log.msg( self.nickname + ": Connection Lost" )
-		if self.logger:
-			self.logger.disconnected()
 
 	def signedOn( self ):
 		self.join( self.factory.channel )
@@ -74,16 +72,26 @@ class GoogleCodeIRCBot ( irc.IRCClient ):
 
 		log.msg( self.nickname + ": Joined " + self.channel )
 		if self.logger:
-			self.logger.joined( channel )
+			self.logger.joined(self.nickname, channel)
 
 		self.lineRate = None
-		self.trysay( "Hello, I'm a Google Code IRC Bot, Version %s" % shared.VERSION )
-		self.trysay( "More details are available at %s" % shared.SOURCE_URL )
+		#self.trysay( "Hello, I'm a Google Code IRC Bot, Version %s" % shared.VERSION )
+		#self.trysay( "More details are available at %s" % shared.SOURCE_URL )
 		self.lineRate = 2
 
+	def userJoined(self, user, channel):
+		if self.logger:
+			self.logger.joined(user, channel)
+
 	def left( self, channel ):
 		log.msg( self.nickname + ": Left Channel " + channel )
 		self.channel = None
+		if self.logger:
+			self.logger.left(self.nickname, channel)
+
+	def userLeft(self, user, channel):
+		if self.logger:
+			self.logger.left(user, channel)
 
 	def trysay( self, msg ):
 		"""
@@ -116,18 +124,23 @@ class GoogleCodeIRCBot ( irc.IRCClient ):
 				return
 
 		if channel == self.channel:
-			if msg.startswith( self.nickname + ":" ):
-				args = msg.split( ' ')
-				if 'open' == args[1].lower():
-					if self.gdata:
-						msg = "%s: There are %d open issues." % ( user, self.gdata.get_issue_count() )
-					else:
-						msg = "Sorry, I have no connection to the issue tracker."
-				elif 'help' == args[1].lower():
-					msg = "%s: Valid commands are [ OPEN, HELP ]. Please visit %s for more help." % ( user, shared.SOURCE_URL )
-				else:
-					msg = "%s: Um, what? Try HELP." % user
-				self.trysay( msg )
+			quotes = [
+				"Life? Don't talk to me about life!",
+				"I have a million ideas. They all point to certain death.",
+				"I could calculate your chance of survival, but you won't like it.",
+				"I'd give you advice, but you wouldn't listen. No one ever does.",
+				"I ache, therefore I am.",
+				"I've seen it. It's rubbish.",
+				"I think you ought to know I'm feeling very depressed.",
+				"Pardon me for breathing, which I never do any way so I don't know why I bother to say it, oh God, I'm so depressed.",
+				"I hate oceans.",
+				"I’m just trying to die.",
+				"I hate oceans.",
+				"Funny, how just when you think life can't possibly get any worse it suddenly does.",
+				"Wearily I sit here, pain and misery my only companions.",
+			]
+			if msg.startswith((self.nickname + ":", self.nickname + ",")):
+				self.trysay("%s: %s" % (user, random.choice(quotes)))
 
 	def action ( self, user, channel, msg ):
 		if self.logger:
diff --git a/gib/logger.py b/gib/logger.py
index 9411991..0c0864e 100644
--- a/gib/logger.py
+++ b/gib/logger.py
@@ -46,14 +46,11 @@ class IRCLogger:
 	def close ( self ):
 		self.file.close()
 
-	def connected ( self ):
-		self.log( "[connected]" )
+	def joined ( self, user, channel ):
+		self.log( "[%s joined %s]" % (user, channel) )
 
-	def disconnected ( self ):
-		self.log( "[disconnected]" )
-
-	def joined ( self, channel ):
-		self.log( "[joined %s]" % channel )
+	def left(self, user, channel):
+		self.log("[%s left %s]" % (user, channel))
 
 	def message ( self, user, message ):
 		self.log( "<%s> %s" % ( user, message ) )
diff --git a/gib/readers.py b/gib/readers.py
index aea31ca..b18f06b 100644
--- a/gib/readers.py
+++ b/gib/readers.py
@@ -85,7 +85,7 @@ class GoogleFeedReader:
 		for entry in feed['entries']:
 			if entry['id'] == self.last_id:
 				break
-			added.append( entry )
+			added.insert(0, entry)
 		if 1 <= len( feed['entries'] ):
 			self.last_id = feed['entries'][0]['id']
 		return added
diff --git a/googlecode-irc-bot.py b/googlecode-irc-bot.py
index 2ae7c92..10d660e 100644
--- a/googlecode-irc-bot.py
+++ b/googlecode-irc-bot.py
@@ -109,17 +109,12 @@ def main ():
 			bot_processes[bot.name].start()
 			print "Started", bot.name
 
-	while True:
+	for key, process in bot_processes.items():
 		try:
-			sleep( 60 ) # TODO: Make configurable
-			for key, process in bot_processes.items():
-				if not process.is_alive():
-					print "Bot", key, "has died!"
-		except KeyboardInterrupt, e:
-			print "Shutting Down"
-			exit()
-		except Exception, e:
-			print "Error:", e
+			process.join()
+		except:
+			pass
+		print "Stopped", key
 
 if __name__ != '__main__':
 	print "This can only be run as a main entry point."
